ambry-router/src/main/java/com/github/ambry/router/DeleteOperation.java

```diff
@@ -379,6 +414,60 @@ private void checkAndMaybeComplete() {
     }
   }
 
+  /**
+   * Try to replicate the blob and then retry the Delete Operation
+   * @return true if it's under retry. return false if cannot do retry or retry is finished.
+   */
+  private boolean retryWithReplicateBlob() {
+    // it's the first time we hit the 503
+    if (replicateBlobCallback == null) {
+      List<ReplicaId> successfulReplica = operationTracker.getSuccessReplica();
+      DataNodeId sourceDataNode = successfulReplica.get(0).getDataNodeId();
```

Comment:
so u assume the node to return success is the first in the list ?

Reply:
We generate one DeleteOperation object per delete request and the replicateBlobCallback is null when the DeleteOperation is created.
It's not in the "node" scope.